components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/SessionMgtUtils.java

```diff
@@ -0,0 +1,224 @@
+     */
+    public static String[] getSQLFiltersFromExpressionNodes(List<ExpressionNode> expressionNodes)
+            throws UserSessionException {
+
+        String sessionFilter = "";
+        String appFilter = "";
+        String userFilter = "";
+        String mainFilter = "";
+        StringJoiner sessionJoiner = new StringJoiner(SessionMgtConstants.AND);
+        StringJoiner appJoiner = new StringJoiner(SessionMgtConstants.AND);
+        StringJoiner userJoiner = new StringJoiner(SessionMgtConstants.AND);
+        StringJoiner mainJoiner = new StringJoiner(SessionMgtConstants.AND);
+
+        for (ExpressionNode expressionNode : expressionNodes) {
+            String operation = expressionNode.getOperation();
+            String value = expressionNode.getValue();
+            String attribute = expressionNode.getAttributeValue();
+            String filterType = "default";
+
+            StringBuilder filter = new StringBuilder();
+            boolean isString = true;
+
+            switch (attribute.toLowerCase()) {
+                case SessionMgtConstants.FLD_SESSION_ID_LOWERCASE:
+                    attribute = SessionMgtConstants.COL_SESSION_ID;
+                    filterType = "session";
+                    break;
+                case SessionMgtConstants.FLD_APPLICATION_LOWERCASE:
+                    attribute = SessionMgtConstants.COL_APPLICATION;
+                    value = value.toLowerCase();
+                    filterType = "app";
+                    break;
+                case SessionMgtConstants.FLD_LOGIN_ID_LOWERCASE:
+                    attribute = SessionMgtConstants.COL_LOGIN_ID;
+                    value = value.toLowerCase();
+                    filterType = "user";
+                    break;
+                case SessionMgtConstants.FLD_IP_ADDRESS_LOWERCASE:
+                    attribute = SessionMgtConstants.COL_IP_ADDRESS;
+                    break;
+                case SessionMgtConstants.FLD_USER_AGENT_LOWERCASE:
+                    attribute = SessionMgtConstants.COL_USER_AGENT;
+                    value = value.toLowerCase();
+                    break;
+                case SessionMgtConstants.FLD_LOGIN_TIME_LOWERCASE:
+                    attribute = SessionMgtConstants.COL_LOGIN_TIME;
+                    break;
+                case SessionMgtConstants.FLD_LAST_ACCESS_TIME_LOWERCASE:
+                    attribute = SessionMgtConstants.COL_LAST_ACCESS_TIME;
+                    break;
+                case SessionMgtConstants.FLD_TIME_CREATED_SINCE:
+                case SessionMgtConstants.FLD_TIME_CREATED_UNTIL:
+                    attribute = SessionMgtConstants.COL_TIME_CREATED;
+                    isString = false;
+                    break;
+                default:
+                    throw new UserSessionException("Invalid filter attribute: " + attribute);
+            }
+
+            switch (operation.toLowerCase()) {
```

Comment:
Do we need operation to be case insensitive?

Reply:
We might have to keep this as it is now since we have already released it in support. If we try to change, could introduce migration issues